irohad/main/impl/consensus_init.hpp

```diff
@@ -69,6 +57,16 @@ namespace iroha {
             std::shared_ptr<shared_model::interface::CommonObjectsFactory>
                 common_objects_factory);
 
+        // coordinator has a worker, and a factory for coordinated
+        // observables, subscribers and schedulable functions.
+        //
+        // A new thread scheduler is created
+        // by calling .create_coordinator().get_scheduler()
+        rxcpp::observe_on_one_worker coordination_{
+            rxcpp::observe_on_new_thread()
+                .create_coordinator()
+                .get_scheduler()};
```

Comment:
It seems we store a scheduler named (and commented) as coordinator. Is it really what we want? Maybe we should either rename the variable to something like `separate_thread_scheduler_` or store a coordinator here and get the scheduler from it where needed.